chain/owasm/execute_test.go

```diff
@@ -82,8 +94,54 @@ func TestAllocateFailWithExceedMemory(t *testing.T) {
 	size := make([]byte, 8)
 	binary.LittleEndian.PutUint64(size, uint64(8500000))
 
-	_, _, errExecute := Execute(&mockExecutionEnvironment{}, code, "execute", size, 100000000000000000)
+	_, _, errExecute := Execute(&mockExecutionEnvironment{
+		maximumResultSize:               1024,
+		maximumCalldataOfDataSourceSize: 1024,
+	}, code, "execute", size, 100000000000000000)
 	require.NotNil(t, errExecute)
 }
 
+func TestExecuteInvalidGetMaximumCalldataOfDataSourceSize(t *testing.T) {
+	code, err := ioutil.ReadFile("./res/silly.wasm")
+	require.Nil(t, err)
+	env := &mockExecutionEnvironment{
+		maximumCalldataOfDataSourceSize: 12,
+	}
+
+	// It should return error because "band-protocol" has length equal to 13.
+	_, _, err = Execute(env, code, "prepare", []byte{}, 10000)
+	require.NotNil(t, err)
+
+	// It should print "RequestExternalData: DataSourceID = 1, ExternalDataID = 1"
+	// and not return error.
+	_, _, err = Execute(&mockExecutionEnvironment{
+		maximumCalldataOfDataSourceSize: 13,
+	}, code, "prepare", []byte{}, 10000)
+	require.Nil(t, err)
+}
+
+func TestExecuteInvalidGetMaximumResultSize(t *testing.T) {
+	code, err := ioutil.ReadFile("./res/silly.wasm")
+	require.Nil(t, err)
+	env := &mockExecutionEnvironment{
+		externalDataResults:               [][][]byte{nil, {[]byte("RETURN_DATA")}},
+		requestExternalDataResultsCounter: [][]int64{nil, []int64{0}},
+		maximumResultSize:                 10,
+	}
+
+	// It should return error because "band-protocol" has length equal to 13.
```

Comment:
Is it correct comment?

Reply:
What 's wrong?

Reply:
`// It should return error because "RETURN_DATA" has length equal to 11.`